src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -403,7 +404,12 @@ private SnapshotProject retrieveSnapshotProject(String sql, MapSqlParameterSourc
         // add the filter to the clause to get the actual items
         DaoUtils.addFilterClause(filter, params, whereClauses);
         String whereSql = "";
-        if (!whereClauses.isEmpty()) {
+        String joinSql = "";
+        if (!whereClauses.isEmpty() || !StringUtils.isEmpty(datasetId)) { //TODO can whereClauses be empty w line 402?
+            if (!StringUtils.isEmpty(datasetId)) {
+                whereClauses.add(" snapshot_source.dataset_id = :datasetId ");
+                joinSql = " JOIN snapshot_source ON snapshot.id = snapshot_source.snapshot_id ";
```

Comment:
The SQL looks fine to me. I usually use implicit joins instead of explicit unless I know performance will be an issue, but I don't think there's any great reason to pick one over the other.

Reply:
Its just syntax in Postgres. The optimizer doesn't care.

This method of constructing the SQL string feels pretty awkward to me, but it is consistent with what we do elsewhere. We have talked about changing to use ST like we do in the BQ PDAO, but that seems like a change you'd want to make as a separate PR.